ocaml/idl/datamodel.ml

```diff
@@ -906,6 +906,10 @@ let _ =
     ~doc:"The pool failed to disable the external authentication of at least one host." ();
   error Api_errors.pool_auth_disable_failed_permission_denied ["host";"message"]
     ~doc:"The pool failed to disable the external authentication of at least one host." ();
+  error Api_errors.pool_joining_host_must_have_same_api_version []
```

Comment:
How about adding the master and joining-host's versions as arguments to the error? Similarly below.